mikupad.html

```diff
@@ -1686,9 +1716,12 @@
 	});
 	if (!res.ok)
 		throw new Error(`HTTP ${res.status}`);
+	window.TTS.new_prompt = options.prompt;
+	window.TTS.new_text = "";
 	for await (const chunk of parseEventStream(res.body)) {
 		const probs = chunk.completion_probabilities[0]?.probs ?? [];
 		const prob = probs.find(p => p.tok_str === chunk.content)?.prob;
+		window.TTS.new_text += chunk.content;

```

Comment:
Please don't add code like this to `llamaCppCompletion`; it ties this functionality to the llama.cpp backend. Instead, add the code to the `predict` method in the `App` component. You could also add a `useEffect` with `[promptText, cancel]` as dependencies, so it gets executed after a prediction is finished.

Reply:
Usin React hooks is a great idea. I'll try that, thanks.

Reply:
Well, it was a good idea, but unfortunately it seems to be very unreliable. Both with promptText and promptChunks, it randomly misses a few tokens, most often the last one.
I'll have to add a few lines to App.predict instead. I'll try to modify it as little as possible, and instead add tts-specific methods to be called from there.